File: algoritmos/Exercicio_2/Animal.php
<?php

class Animal{

    public $peso;
    public $patas;

    public function __construct($peso, $patas){

        $this->peso = $peso;
        $this->patas=$patas;
        
    }

    public function comer(){
        return "estou comendo";
    }

    public function andar(){

        return "e logo vou andar!";

    }

}

?>

File: algoritmos/desafioPOO/desafioPOO/src/desafiopoo/model/Animal.java
package desafiopoo.model;

public abstract class Animal {
    // atributos
    private float peso;
    private int qntPatas;
    
    // construtor
    public Animal(float peso, int qntPatas) {
        this.peso = peso;
        this.qntPatas = qntPatas;
    }

    // getters e setters
    public float getPeso() {
        return peso;
    }

    public void setPeso(float peso) {
        this.peso = peso;
    }

    public int getQntPatas() {
        return qntPatas;
    }

    public void setQntPatas(int qntPatas) {
        this.qntPatas = qntPatas;
    }
    
    // demais metodos
    public String andar() {
        return "estou andando.";
    }
    public String comer() {
        return "estou comendo.";
    }
}

